katerynamelnykova/fandom-cookbook-server
# Added the router and a MongoDB Docker container

# What does this PR do?
1. Adds a router
2. Adds a MongoDB Docker container.
3. Adds endpoints

# How to test this PR:
1. To build the image run:
```
make db_build
```
or:
```
docker build -f Dockerfile -t fandom-cookbook/mongodb .
```
2. To start the database run:
```
make docker_run
```
or:
```
docker run -d -p 27017:27017 --name fandom-cookbook_mongodb fandom-cookbook/mongodb
```
4. To restore the db run:
```
docker exec -it fandom-cookbook_mongodb bash
```
```
mongorestore dbbackup/
```
5. To check if the database contains the required data, run the following commands inside the Docker container:
```
mongosh
```
```
use fandom-cookbook-db
```
```
db.books.find()
```
6. To test the backend run:
```
go run main.go
```
```
curl http://localhost:8080/health
```
```
curl http://localhost:8080/get-short-fandom-info
```
```
curl http://localhost:8080/book/654ea9dbdb818788964fa30e
```
```
curl http://localhost:8080/book/654ea9dbdb818788964fa30e/recipe/654eab93db818788964fa312
```
